src/main/java/org/jetbrains/bsp/bazel/server/bazel/data/SemanticVersion.java

```diff
@@ -0,0 +1,79 @@
+package org.jetbrains.bsp.bazel.server.bazel.data;
+
+public class SemanticVersion implements Comparable<SemanticVersion> {
+
+  private static final String SEMANTIC_VERSION_REGEX = "[0-9]+(\\.[0-9]+)*";
+  private static final int MAJOR_VERSION_LOCATION = 0;
+  private static final int MINOR_VERSION_LOCATION = 1;
+  private static final int PATCH_VERSION_LOCATION = 1;
+
+  private final String version;
+  private final String[] versionParts;
+
+  public SemanticVersion(String version) {
+    if (version == null) {
+      throw new IllegalArgumentException("Version can not be null");
+    }
+    if (!version.matches(SEMANTIC_VERSION_REGEX)) {
+      throw new IllegalArgumentException("Invalid version format");
+    }
+
+    String[] parts = version.split("\\.");
+    if (parts.length != 3) {
+      throw new IllegalArgumentException("Invalid number of version parts");
+    }
+
+    this.version = version;
+    this.versionParts = parts;
+  }
+
+  public final int getMajorVersion() {
+    return Integer.parseInt(versionParts[MAJOR_VERSION_LOCATION]);
```

Comment:
it would be better to do the parsing in the constructor and simply return the value here